### ISSUE TITLE
energy conservation problem (Ubal ~ -0.1 even without plasma) 

### ISSUE THREAD

**Issue by aeriforme** [SmileiPIC/Smilei#40]
Hello everyone!

I'm experiencing some issues with the "Ubal" diagnostic, which persist even for the simplest simulations (i.e. without a plasma). 

Using this input file: 

``` python
import math
l0 = 2.0*math.pi 
Lx = 102.4*l0 
Ly = 204.8*l0
Tsim = 200.*t0 
resx = 50.
dx = l0/resx;
dy = l0/resx; 
dt = 0.98 /  math.sqrt( dx**(-2) + dy**(-2) )
rest = int(l0/dt) 
t0 = l0
Main(
    geometry = "2Dcartesian",
    interpolation_order = 2,
    timestep = dt,
    simulation_time = Tsim,
    cell_length  = [dx, dy],
    grid_length = [Lx, Ly],
    number_of_patches = [8,1024],
    EM_boundary_conditions = [
        ["silver-muller","silver-muller"],
        ["silver-muller","silver-muller"],
    ],
    print_every = 100.0,
    random_seed = smilei_mpi_rank
)
LaserGaussian2D(
    box_side        = "xmin",
    a0              = 2.,
    focus           = [20.*l0, Main.grid_length[1]/2.],
    waist           = 5.*l0,
    time_envelope   = tgaussian(start=0., duration=45.0*l0, fwhm=15.*l0, center=22.5*l0, order=2)
)
DiagScalar(
    every = 1.
)
```

Ubal oscillates around -0.1, which should mean that 10% of the laser energy is lost, if I've correctly understood the documentation. Am I doing something wrong? 
Any help would be greatly appreciated! 

Best regards

Arianna

**Comment by mccoys** [comment]
The Maxwell solver is clearly not the problem because your case shows a stable energy in the box once the laser has fully entered the box. Instead, it looks like a diagnostic problem, likely in the computation of the Poynting flux. We will investigate as soon as possible.

Have you tried using only the ``Utot`` scalar ? It should eventually equal the total laser energy.

**Comment by MickaelGrech** [comment]
Hello,
So, we had a look at your input file which looks fine.
The simulation results do look correct and you indeed have energy conservation.
The problem is in the diagnostic and appears when the laser is entering the box.
We will have a closer look at the diagnostic to compute the energy balance when the laser is entering or exiting the box.
Thanks for the feedback!

**Comment by mccoys** [comment]
@aeriforme Sorry for the very long delay. We have found a simple mistake in the way the Poynting flux is computed. It will be fixed soon.

**Comment by mccoys** [comment]
This has been fixed since 5e26d846
